src/modules/backtest.js

```diff
@@ -145,4 +147,137 @@ module.exports = class Backtest {
       });
     });
   }
+
+  getBacktestSummary(signals) {
+    return new Promise(async resolve => {
+
+      let initialCapital = 10000;           //1000 $ Initial Capital
+      let workingCapital = initialCapital;  //Capital that changes after every trade
+      
+      let lastPosition;                     //Holds Info about last action
+      
+      let averagePNLPercent = 0;            //Average ROI or PNL Percentage
+
+      let trades = {                        
+        profitableCount: 0,                 //Number of Profitable Trades
+        lossMakingCount: 0,                 //Number of Trades that caused a loss
+        total: 0,                           //Totol number of Trades
+        profitabilityPercent: 0             //Percentage Of Trades that were profitable
+      }
+
+      let cumulativePNLPercent = 0;         //Sum of all the PNL Percentages
+      let pnlRateArray = [];                //Array of all PNL Percentages of all the trades
+      
+      //Iterate over all the signals
+      for (let s = 0; s < signals.length; s++) {
```

Comment:
I am using for loop to maintain the chronological order of the signal sequence because trades are opened and closed alternatively always squaring off the previous trade. 

forEach does not guarantee the order of iteration and cause the squaring off of the wrong trades.

Reply:
ok interesting, never thought about that.
Is the chronological order guaranteed in js arrays anyway though?

Reply:
In for loop, you can access them serially by incrementing one index at a time, ensuring that you are accessing the signals in chronological order.